lips/lip-4.md

```diff
@@ -0,0 +1,222 @@
+---
+lip: 4
+title: Transaction Metadata Specification
+author: Kevin Hurley (@kphfb)
+status: Draft
+type: Informational
+created: 06/26/2020
+---
+# Summary
+---
+Merchants and wallet holders may rely on third parties to maintain custody of their wallet and may optionally want to receive additional data in transactions.  Libra leverages subaddressing to provide this functionality.
+
+---
+# Terminology
+___
+*subaddress:* Accounts on-chain are represented by an address.  To allow multiplexing of a single address into multiple wallets, custodial wallets may utilize “subaddresses” under the on-chain address for each underlying user.  These subaddresses represent the users and may have meaning only to the custodian - but the mapping from subaddress to user is not known on-chain, but rather is an internal mapping known by the custodian.
+*referenced_event:* In the case where funds must be returned, referenced_event refers to the event sequence number of the sender’s original sent payment event.  Since refunds are just another form of p2p transfer, the referenced event field allows a refunded payment to refer back to the original payment.
+
+
+---
+# Abstract / Motivation
+---
+
+Merchants and wallet holders may rely on third parties to maintain custody of their wallet and may optionally want to receive additional data in transactions.  Libra leverages subaddressing to provide this functionality.  This document describes an approach to supporting transactions like this by defining a standard that supports:
+
+* Subaddressing to support multiplexing a single account address into multiple wallets, enabling custodial wallets to maintain a single address rather than maintain numerous addresses.
+
+As an example, assume a recipient account provides custodial wallets. To send funds to this wallet within Libra:
+
+* The recipient crafts a QR code containing a URI that references the account hosting the custodial wallet, or the host, and a reference to their wallet.
```

Comment:
Might be clearer to avoid introducing the term "host" and just keep saying "custodial wallet" or "custodian"